consensus/src/liveness/leader_reputation.rs

```diff
@@ -499,22 +503,17 @@ impl ProposerElection for LeaderReputation {
             .heuristic
             .get_weights(self.epoch, &self.proposers, &sliding_window);
         assert_eq!(weights.len(), self.proposers.len());
-        let mut total_weight = 0;
-        for w in &mut weights {
-            total_weight += *w;
-            *w = total_weight;
-        }
-        let mut state = round.to_le_bytes().to_vec();
-        let chosen_weight = next(&mut state) % total_weight;
-        let chosen_index = weights
-            .binary_search_by(|w| {
-                if *w <= chosen_weight {
-                    Ordering::Less
-                } else {
-                    Ordering::Greater
-                }
-            })
-            .unwrap_err();
+        // Multiply weights by voting power:
+        weights
```

Comment:
Are the other weights (beside stake which is added here) just proposer performance?

Reply:
yeah this is just selection from:

proposer_and_voter_config.active_weight,
proposer_and_voter_config.inactive_weight,
proposer_and_voter_config.failed_weight,

Reply:
What are active and inactive weights? Also does it make sense to multiply this existing weight by stake instead of adding?

Reply:
defaults:

                   active_weight: 1000,
                    inactive_weight: 10,
                    failed_weight: 1,
                    
 and yes, we should be multiplying, and we are :) we use *= few lines below

Reply:
What are proposer_and_voter_config.active_weight and proposer_and_voter_config.inactive_weight and how are they calculated?